### PR TITLE
Endpoint dev

### PR DESCRIPTION
**Pull request by HadarGreinsmark**
Working version of authorize endpoint.

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -5,5 +5,6 @@ go:
 
 before_install:
  - go get github.com/streadway/amqp
+ - go get github.com/stretchr/testify
 script:
  - go build
```
Comment (dskeppstedt) [inline .travis.yml]: add go test

File: apiutil/endpoints.go [hunk 2]
```diff
@@ -0,0 +1,102 @@
...
+		return
+	}
+
+	if amqpRes.Code != 0 {
+		w.WriteHeader(http.StatusUnauthorized)
+		return
+	}
+
+
+	// TODO: Use data from amqpResponse to send to client
```
Comment (dskeppstedt) [inline apiutil/endpoints.go]: Remove

File: apiutil/endpoints.go [hunk 3]
```diff
@@ -0,0 +1,102 @@
...
+
+	amqpRes := amqpAuthorizeResponse{}
+	err = json.Unmarshal(amqpResponse, &amqpRes)
+	if err != nil {
+		w.WriteHeader(http.StatusInternalServerError)
+		log.Println(err.Error())
+		return
+	}
+
+	if amqpRes.Code != 0 {
```
Comment (dskeppstedt) [inline apiutil/endpoints.go]: There needs to be a difference between error and not authorized

File: apiutil/endpoints.go [hunk 4]
```diff
@@ -0,0 +1,102 @@
...
+		Value:   amqpRes.SessionToken,
+		Expires: time.Now().AddDate(1, 0, 0), // One year ahead
+	})
+
+	w.Header().Set("Access-Control-Allow-Origin", "*")
+	fmt.Fprint(w, )
+}
+
+func Stats(w http.ResponseWriter, r *http.Request, rpc RPCaller) {
+	println("stttta")
```
Comment (dskeppstedt) [inline apiutil/endpoints.go]: Remove

File: apiutil/endpoints.go (hunk 3 above)
Comment (dskeppstedt) [reply apiutil/endpoints.go]: Also add another test

File: apiutil/rpc.go [hunk 5]
```diff
@@ -0,0 +1,71 @@
...
+
+	// Wait for incoming AMQP messages, then forward the body to the requester
+	consumer, err := amqpChannel.Consume(queue.Name, "", false, true, false, false, nil)
+	CheckError(err)
+	go func() {
+		for msg := range consumer {
+			answered, ok := qm.expectedResponses[msg.CorrelationId]
+
+			if !ok {
+				// TODO: use logger instead
```
Comment (dskeppstedt) [inline apiutil/rpc.go]: fix this todo

File: pajj_api.go [hunk 6]
```diff
@@ -1,174 +1,38 @@
...
-		delete(qm.expectedResponses, qm.amqpQueue.Name)
-		checkError(err)
-	}
-	return respondChannel
-}
-
 func main() {
-	
-	log.Println("Starting the server")
+	print("starting server")
```
Comment (dskeppstedt) [inline pajj_api.go]: nooo, bring back log.Println

**Review by dskeppstedt — CHANGES_REQUESTED** [review]
Looks good, some changes needed.
Maybe change from apiutil to api.

**Comment by HadarGreinsmark** [comment]
@dskeppstedt, the requested changes are done!

**Review by dskeppstedt — APPROVED** [review]
Good job!

**Pull request merged**